src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -135,6 +139,7 @@
 
     private static class FactoryMapHolder {
         static final Map<String, Provider<? extends RandomGenerator>> FACTORY_MAP = createFactoryMap();
```

Comment:
Unrelated to this PR, but a more general question. It appears that this `FACTORY_MAP` gets cached on first use/call. A few questions about the `createFactoryMap` method:
1. The javadoc of that private method says:
```
/**
         * Returns the factory map, lazily constructing map on first use.
         *
         * @return Map of RandomGeneratorFactory classes.
         */
```
But the implementation and the signature of this method actually return a Map of `RandomGenerator` classes and not the `RandomGeneratorFactory` classes.
2. The implementation of this method internally uses the `ServiceLoader` to load the `RandomGenerator` service provider implementations. The internal implementation of the `ServiceLoader` will use a Thread context classloader that is set on the calling thread. The result of the call to this `createFactoryMap` is then cached once and for all in the `FACTORY_MAP`. Would this then lead to a non-deterministic behaviour where whoever ends up initializing this `FactoryMapHolder` first, will end up storing those RandomGenerators for every one else? Is this intentional? Do you think this caching should be reviewed?

Reply:
Good point. It might be useful to explicitly pass the boot layer to the service loader.

But that is outside the scope of this bug - my goal here is just to make it not throw an exception when running with a SecurityManager while not introducing security vulnerabilities.

Reply:
> But that is outside the scope of this bug

Agreed. I'm not expecting that to be changed in this PR :)  It was more a general question about the implementation.